client/src/explorer/walletExplorer.ts

```diff
@@ -71,48 +71,48 @@ export class BlockchainWalletExplorerProvider implements BlockchainExplorerProvi
+        for (const walletRegistryEntry of allWalletRegistryEntires) {
 
             if (walletRegistryEntry.walletPath) {
                 // get identityNames in the wallet
                 const walletGenerator: IFabricWalletGenerator = FabricWalletGeneratorFactory.createFabricWalletGenerator();
                 const wallet: IFabricWallet = walletGenerator.getNewWallet(walletRegistryEntry.walletPath);
-                identityNames = await wallet.getIdentityNames();
+                const identityNames: string[] = await wallet.getIdentityNames();
 
-                treeState = identityNames.length > 0 ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.None;
+                const treeState: vscode.TreeItemCollapsibleState = identityNames.length > 0 ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.None;
 
-                tree.push(new WalletTreeItem(this, walletRegistryEntry.name, identityNames, treeState));
+                if (walletRegistryEntry.managedWallet) {
+                    tree.push(new LocalWalletTreeItem(this, walletRegistryEntry.name, identityNames, treeState));
+                } else {
+                    tree.push(new WalletTreeItem(this, walletRegistryEntry.name, identityNames, treeState));
+                }
             }
         }
 
         return tree;
     }
 
-    private async createIdentityTree(walletTreeItem: WalletTreeItem): Promise<BlockchainTreeItem[]> {
+    private async createIdentityTree(walletTreeItem: WalletTreeItem | LocalWalletTreeItem): Promise<BlockchainTreeItem[]> {
         const tree: Array<BlockchainTreeItem> = [];
 
         // Populate the tree with the identity names
+        const walletName: string = walletTreeItem.name;
         for (const identityName of walletTreeItem.identities) {
-            if (walletTreeItem.name === FabricWalletUtil.LOCAL_WALLET && identityName === FabricRuntimeUtil.ADMIN_USER) {
-                // Create Local Fabric identity tree item - user cannot delete this
-                tree.push(new AdminIdentityTreeItem(this, identityName, walletTreeItem.name));
+            if (walletTreeItem instanceof LocalWalletTreeItem) {
+                const adminIdentities: FabricIdentity[] = await FabricRuntimeManager.instance().getRuntime().getIdentities(walletName);
+                const isAdminIdentity: boolean = adminIdentities.some((adminIdentity: FabricIdentity): boolean => adminIdentity.name === identityName);
+                if (isAdminIdentity) {
+                    // User can't delete this!
+                    tree.push(new AdminIdentityTreeItem(this, identityName, walletName));
+                } else {
+                    tree.push(new IdentityTreeItem(this, identityName, walletName));
+                }
             } else {
-                tree.push(new IdentityTreeItem(this, identityName, walletTreeItem.name));
+                tree.push(new IdentityTreeItem(this, identityName, walletName));
```

Comment:
(review ongoing) This looks the same as line 112 - do you need both?

Reply:
@heatherlp I suppose not - rewrote the block so that the duplicate line does not occur.